pkgs/development/tools/misc/libtool/libtool2.nix

```diff
@@ -36,6 +37,25 @@ stdenv.mkDerivation rec {
     popd
   '';
 
+  # When detecting the `LD` (linker) command and flags for cross
+  # compiling, libtool.m4 expects to be able to use "/usr/bin/file" to
+  # detect host details like ABI and linker flags.  Unfortunately this
+  # detection code hardwires the path to /usr/bin/file, which is
+  # inaccessible for sandboxed builds.  This causes the `LD` flag to
+  # be detected incorrectly when cross-compiling to the following
+  # hostPlatforms: {x86_64, powerpc, s390, sparc, mips64*}-linux,
+  # x86_64-kfreebsd, *-solaris, *-irix, and *-hpux.
+  #
+  # This substitution is performed only for cross-compilation in order
+  # to avoid a mass-rebuild.  A separate PR will be submitted to
+  # staging, not to be merged until after nixpkgs-22.05 branch-off,
+  # which deletes this conditional and this comment.
+  #
+  preBuild = if stdenv.buildPlatform == stdenv.targetPlatform
```

Comment:
I would appreciate a review of this line in particular.  I think of `libtool` as being a "compiler-like" package that has a `targetPlatform` but I'm not sure if nixpkgs thinks of it this way.  Please let me know if I should change this from build==target to build==host.